src/Notes/NavigationNudge.php

```diff
@@ -33,13 +33,38 @@ public function __construct() {
 		add_action( 'update_option_' . Navigation::TOGGLE_OPTION_NAME, array( $this, 'action_note' ), 10, 2 );
 	}
 
+	/**
+	 * Should this note exist? (The navigation feature should exist.)
+	 */
+	public static function should_note_exist() {
+		return Features::exists( 'navigation' );
+	}
+
+	/**
+	 * Delete this note if the navigation feature does not exist, unless the note has been soft-deleted already.
+	 */
+	public static function delete_if_not_supported() {
```

Comment:
Whatever we go with naming-wise above, I think we should mirror here.  `delete_if_not_applicable`.

And I think it would be great to move this to `NoteTraits` as you suggested.